src/lib.rs

```diff
@@ -99,6 +99,101 @@ impl MetalIrConverter {
     }
 }
 
+macro_rules! verioned_info {
+    ($name:ident, $create:ident, $release:ident) => {
+        pub struct $name {
+            me: ffi::$name,
+            funcs: Arc<bindings::metal_irconverter>,
+        }
+
+        impl Deref for $name {
+            type Target = ffi::$name;
+
+            fn deref(&self) -> &Self::Target {
+                &self.me
+            }
+        }
+
+        impl Drop for $name {
+            fn drop(&mut self) {
+                assert!(unsafe { self.funcs.$release(&mut self.me) })
+            }
+        }
+
+        impl $name {
+            unsafe fn new(
+                reflection: &IRShaderReflection,
+                version: ffi::IRReflectionVersion,
+            ) -> Option<Self> {
+                let mut info = MaybeUninit::uninit();
+                let success =
+                    reflection
+                        .funcs
+                        .$create(reflection.me.as_ptr(), version, info.as_mut_ptr());
+
+                success.then(|| Self {
```

Comment:
Nit: we usually prefer regular if statements in the code we maintain for readability.

Reply:
Fixed in [da3b23e](https://github.com/Traverse-Research/saxaboom/pull/38/commits/da3b23e0c558a5ca5001cc8f0053d342d312d583)